### PR TITLE
Replaces intents with combat mode

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

This removes intents and replaces them with combat mode and alternate right-click action. Default keybinds are 4 to enable combat mode and 1 to disable, same as previous harm/help intents.

Major changes:
- Removes disarm and grab intent, help/harm intents are now combat mode off/on
- Right-click menu moved to shift + right-click
- Right click is now disarm, grabbing is still ctrl-click, both are possible with or without combat mode enabled
- Default movement lock key is now alt, to allow grabbing while moving (this does not change existing save files)
- Some martial arts have been reworked, in-game instructions have been updated to reflect changes

Interaction changes:
- Right click to deconstruct tables/racks/walls/etc, and to open the wire panel of autolathes
- Right click to harm when using a baton or bo staff
- Right click to unlock APCs and air alarms, alt-click to discharge energy into them as an ethereal or recharge as a preternis.
- Right click to malpractice surgery
- Right click to hold people at gunpoint
- Right click to use PDA while in ID slot
- Right click to open a storage item (on top of alt click)
- Right click to fire kinetic crusher projectile, left click can now cleave attack with combat mode on
- Right click to toggle pumps and other atmospherics machines

Plus more that I likely forgot about. In general if something used to be done with disarm or grab intent, it's probably done with right-click or ctrl-click now.

Original PR here: tgstation/tgstation#56601

# Why is this good for the game?

After having been on other servers that use the combat mode system, it's difficult to go back to using intents because of how clunky and unintuitive they feel compared to it.

To anyone who's already used to it and don't know anything else, intents might seem intuitive on the surface in a "what do you want to do with your click" sort of way, but this goes out the window when clicking on anything that isn't another player where "disarm" and "grab" don't make much sense, like when deconstructing a table.

As the original PR puts it:

"Intents heavily cripple both the code and the UI design of interactions. While I understand that a lot of people will dislike this PR as they are used to intents, they are one of our weakest links in terms of explaining to players how to do specific things, and require a lot more keypresses to do compared to this.

As an example, martial arts can now be done without having to juggle 1 2 3 and 4 to switch intents quickly.

The full intention of this shift in control scheme is that right click will become 'secondary interaction' for items, which prevents some of the awkward juggling we have now with item modes etcetera."

# Testing

It's like 300+ files changed, no way I can test every single thing I've touched, but everything I've tested so far does work.

Buster arm demonstration:

https://github.com/yogstation13/Yogstation/assets/93578146/2937f24c-3047-4104-9fcf-f7691360be44

# Spriting
image

image

image

image

image

image

image

image

image

# Changelog

:cl:  Qustinnus, SapphicOverload, Molti
experimental: Replaced intents with combat mode
imageadd: Adds combat mode buttons to the UI, replacing intents
bugfix: Fixes some of the martial arts not working sometimes
/:cl:

### CODE REVIEW

**Comment by ShoziOffical** [comment]
combat off icon button is invisible

**Comment by Addust** [comment]
i am neutral on this, but PLEASE give people the option to use intents

**Comment by Therandomhoboo** [comment]
Yes please for the love of god, i dont care if im banned this is just QoL

**Comment by iamthesilly** [comment]
> combat off icon button is invisible

Should be fixed now

**Comment by Cartlord** [comment]
please do not do this

**Comment by Cartlord** [comment]
To elaborate to an extent, while I do understand the usefulness of combat mode and right-click mechanics for things like deconstructing tables, the main thing I'm concerned about is putting the right-click menu - which is still used for quite a few things, as far as I recall - behind shift-right-click. I just don't like the idea of having another keypress for something so common.
I do wish there was another way to handle alternate interactions while keeping right-click unchanged, but I think every hold-a-button-and-left-click combination's taken...

**Comment by ShoziOffical** [comment]
> > combat off icon button is invisible
> 
> Should be fixed now

not 
image
clockwork one is still broken

**Comment by Reed0506** [comment]
combat mode is probably a good evolution of the intent system that generally improves gameplay. the only issue is that transitioning from intents to combat mode is a bit confusing (mainly due to shift right click) but thats not a big deal at the end of the day.

**Comment by Yarinoi** [comment]
i dont think you can ctrl+click pda's anymore to turn them on with combat mode on or off, please fix

**Comment by missatessatessy** [comment]
Based on a short discussion on discord I believe that replacing people's natural instinct to press right click with having to use shift + right click isn't a great implementation. Perhaps an alternative can be found so that right click can remain the same?

**Comment by MommaDragon** [comment]
Although she hasn't actually tried the system out yet, it might've been a much better idea to ask about the idea of doing this system (or the idea of the system alone) rather than test-merging it, confusing people and potentially getting negative criticism and causing people to leave.

Instead of going for the finish line and having this randomly merge, only to be berated with insults and given broken, old trophies, asking opinions about the idea or concept of the new combat system would be more helpful - and at that point, you can share more information about what you're planning to do, and if people don't like it, they can tell you why without jumping to being very angry.

Especially now with how people have to SHIFT-right click to open the little menu and its flyouts, which makes basic instincts freak out because right click no longer works as they thought it once did - and now you also have no pressing 1, 2, 3, 4 for combos as normal with martial arts, CQC, Mag Krava...

Everything considered, this just messes with peoples' brains and, even if it would be difficult, making this a togglable setting that starts, by default, disabled, and could be updated at some point (reconnect, ghost goes into a new body, join the game, or server restart?), would most likely make this more cohesive and more opt-in rather than an opt out system, or, god forbid she says this, a very forced system that nobody likes and people leave over. Some people might as well have jumped the gun and assumed that it was going to be a permanent change rather than just a simple testmerge - and with how Deadly's said all he's done is share a link to those MHelps asking about the right click menu (please don't do this, actually talk to people), it most likely *has* made it feel like an unwelcoming mess.

TL;DR: A change that a lot of people neither expected, nor liked, getting dropped on people without asking for pre-project criticism, with people being unhelpful in both regards (people leaving *and* staff not being communicative other than just links or lack of responses), can end up becoming a disaster for those very used to the old system - equally so for new players as now they have yet another thing to learn that isn't easy to understand buttons and their names of what they do.

**Comment by ToasterBiome** [comment]
> Although she hasn't actually tried the system out yet, it might've been a much better idea to ask about the idea of doing this system (or the idea of the system alone) rather than test-merging it, confusing people and potentially getting negative criticism and causing people to leave.
> 
> Instead of going for the finish line and having this randomly merge, only to be berated with insults and given broken, old trophies, asking opinions about the idea or concept of the new combat system would be more helpful - and at that point, you can share more information about what you're planning to do, and if people don't like it, they can tell you why without jumping to being very angry.
> 
> Especially now with how people have to SHIFT-right click to open the little menu and its flyouts, which makes basic instincts freak out because right click no longer works as they thought it once did - and now you also have no pressing 1, 2, 3, 4 for combos as normal with martial arts, CQC, Mag Krava...
> 
> Everything considered, this just messes with peoples' brains and, even if it would be difficult, making this a togglable setting that starts, by default, disabled, and could be updated at some point (reconnect, ghost goes into a new body, join the game, or server restart?), would most likely make this more cohesive and more opt-in rather than an opt out system, or, god forbid she says this, a very forced system that nobody likes and people leave over. Some people might as well have jumped the gun and assumed that it was going to be a permanent change rather than just a simple testmerge - and with how Deadly's said all he's done is share a link to those MHelps asking about the right click menu (please don't do this, actually talk to people), it most likely _has_ made it feel like an unwelcoming mess.
> 
> TL;DR: A change that a lot of people neither expected, nor liked, getting dropped on people without asking for pre-project criticism, with people being unhelpful in both regards (people leaving _and_ staff not being communicative other than just links or lack of responses), can end up becoming a disaster for those very used to the old system - equally so for new players as now they have yet another thing to learn that isn't easy to understand buttons and their names of what they do.

This is a very long form way to say that you personally don’t like the system. Threatening players leaving isn’t going to change the minds of people who think this is a good mechanical change. Also, TG and monke both use combat mode… so unless you’re coming from goon, they’re the same.

**Comment by ShadowDeath6** [comment]
I personally dislike this entire concept just because it feels like change for the sake of change and serves no real quality of life purpose past making martial arts slightly easier to do. I find it a lot more intuitive to have disarm/grab intents than to have specific mouse-clicks for them (plus, the right click menu showing with shift right click is just annoying and very unintuitive compared to right clicking now). It’s not easier or more difficult, just different- and I don’t think the novelty of it being different than what we have now warrants its implementation.

**Comment by iamthesilly** [comment]
> I find it a lot more intuitive to have disarm/grab intents than to have specific mouse-clicks for them

Tell me, which intent opens the panel on an autolathe? Without looking at the code.

With the combat mode system, right click generally deconstructs things rather than use it or damage it, which is much more intuitive.

**Comment by ShadowDeath6** [comment]
> Tell me, which intent opens the panel on an autolathe? Without looking at the code.

I'm going to assume that it's something other than help since that might insert the screwdriver in. I don't hack autolathes enough to remember, but I don't find losing one screwdriver to be anything past a funny "whoops!" moment, and the fact that this PR fixes that is... good? There are other ways to fix that issue than to implement such a wide-reaching change as overhauling the item interaction system.

**Comment by iamthesilly** [comment]
> I'm going to assume that it's something other than help since that might insert the screwdriver in. I don't hack autolathes enough to remember, but I don't find losing one screwdriver to be anything past a funny "whoops!" moment, and the fact that this PR fixes that is... good? There are other ways to fix that issue than to implement such a wide-reaching change as overhauling the item interaction system.

It's not JUST about the autolathe. It's not JUST about martial arts. It's not JUST about deconstructing things. It's that GENERALLY, having an alternate right click action to do things is easier to do and a more intuitive system because "disarm" and "grab" make no sense when interacting with anything that isn't a player. It just doesn't make sense to "disarm" a table.

**Comment by ShadowDeath6** [comment]
> It's not JUST about the autolathe. It's not JUST about martial arts. It's not JUST about deconstructing things. It's that GENERALLY, having an alternate right click action to do things is easier to do and a more intuitive system because "disarm" and "grab" make no sense when interacting with anything that isn't a player. It just doesn't make sense to "disarm" a table.

I don't think it's a very large hurdle for players to surmount to understand that having help intent enabled will put their screwdriver on the table. I don't think we're going to see eye-to-eye on this regardless: I'm sure that players can adapt to it eventually, but I personally dislike this change just because it reworks a system I've gotten quite used to over the however-many years I've played SS13 while changing basically nothing. I didn't like when /tg/ did it because it was just inconvenient to re-learn that left clicking things sometimes didn't work anymore and it didn't add any noticeable QoL. 
I would ask for it to be an opt-in feature so everyone's happy, but I question the feasibility of implementing an opt-in system for overhauling mouse click binds. 
Either way, I don't have anything left to say on this PR past the fact that it feels like change for the sake of change. It replaces the learning curve for intent item interactions to the learning curve of whatever modifier-click you need to use to interact with an item or a player.

**Comment by Cartlord** [comment]
> This is a very long form way to say that you personally don’t like the system. Threatening players leaving isn’t going to change the minds of people who think this is a good mechanical change. Also, TG and monke both use combat mode… so unless you’re coming from goon, they’re the same.

I think it's disingenuous to act as though that's all that was said and ignore statements like "Deadly, in MHelps, is just linking people to this P.R. instead of explaining the changes to right-clicks" and "Maybe you should have asked if people were interested before you made a massive change like this".
This is probably a stupid idea, but that's never stopped us-
What about just removing every intent-based object interaction? Make every intent with screwdrivers on lathes open the panel, because there're almost no circumstances where people recycle those anyways and disposals end in a recycler. Make every intent with a wrench on a table deconstruct it, because barely anyone bothers putting those on tables and throwing works to get them on there anyway.

**Comment by DeadlyBag** [comment]
> > This is a very long form way to say that you personally don’t like the system. Threatening players leaving isn’t going to change the minds of people who think this is a good mechanical change. Also, TG and monke both use combat mode… so unless you’re coming from goon, they’re the same.
> 
> I think it's disingenuous to act as though that's all that was said and ignore things like "Deadly, a mentor, is just linking people to this P.R. instead of explaining the changes to right-clicks" and "Maybe you should have asked if people were interested before you made a massive change like this". This is probably a stupid idea, but that's never stopped us- What about just removing every intent-based object interaction? Make every intent with screwdrivers on lathes open the panel, because there're almost no circumstances where people recycle those anyways and disposals end in a recycler. Make every intent with a wrench on a table deconstruct it, because barely anyone bothers putting those on tables and throwing works to get them on there anyway.

Hey, deadly here, i dont know where this rumor came from where im wordlessly linking PRs in Mhelps. (Haven't answered any mhelps in a while) im also an admin and not a mentor. This never happend and i'd appreciate if my name wasn't used to lie in order to prove a point.

**Comment by JohnFulpWillard** [comment]
I thought we learned from mood that "have it be a pref that is off by default so players ignore their entire existence" was a bad thing.

**Comment by Cartlord** [comment]
> Hey, deadly here, i dont know where this rumor came from where im wordlessly linking PRs in Mhelps. (Haven't answered any mhelps in a while) im also an admin and not a mentor. This never happend and i'd appreciate if my name wasn't used to lie in order to prove a point.


> [...]and with how Deadly's said all he's done is share a link to those MHelps asking about the right click menu (please don't do this, actually talk to people), it most likely _has_ made it feel like an unwelcoming mess.

@MommaDragon please explain, you were the one to make the claim so i expect you to have something to back it up
Also, Deadly, I apologize for referring to you as a Mentor - I really don't have an excuse for getting that wrong. I have made slight edits to my post.

**Comment by Addust** [comment]
you fucked right click pickup you fool

**Comment by iamthesilly** [comment]
> you fucked right click pickup you fool

If you mean the right click menu, that's shift + right click now.

**Comment by MommaDragon** [comment]
> > Hey, deadly here, i dont know where this rumor came from where im wordlessly linking PRs in Mhelps. (Haven't answered any mhelps in a while) im also an admin and not a mentor. This never happend and i'd appreciate if my name wasn't used to lie in order to prove a point.
> 
> > [...]and with how Deadly's said all he's done is share a link to those MHelps asking about the right click menu (please don't do this, actually talk to people), it most likely _has_ made it feel like an unwelcoming mess.
> 
> @MommaDragon please explain, you were the one to make the claim so i expect you to have something to back it up Also, Deadly, I apologize for referring to you as a Mentor - I really don't have an excuse for getting that wrong. I have made slight edits to my post.

to be honest that was probably a bit exaggerated but then again he said this, which didn't really help her responses.
image
And for some reason (though he was at least willing to clear up that he didn't intend to), he sounded dismissive of the situation.
Alongside how it was suddenly dropped on people, which was both confusing *and* made people not quite happy with Yogstation.

**Comment by MommaDragon** [comment]
> > Although she hasn't actually tried the system out yet, it might've been a much better idea to ask about the idea of doing this system (or the idea of the system alone) rather than test-merging it, confusing people and potentially getting negative criticism and causing people to leave.
> > Instead of going for the finish line and having this randomly merge, only to be berated with insults and given broken, old trophies, asking opinions about the idea or concept of the new combat system would be more helpful - and at that point, you can share more information about what you're planning to do, and if people don't like it, they can tell you why without jumping to being very angry.
> > Especially now with how people have to SHIFT-right click to open the little menu and its flyouts, which makes basic instincts freak out because right click no longer works as they thought it once did - and now you also have no pressing 1, 2, 3, 4 for combos as normal with martial arts, CQC, Mag Krava...
> > Everything considered, this just messes with peoples' brains and, even if it would be difficult, making this a togglable setting that starts, by default, disabled, and could be updated at some point (reconnect, ghost goes into a new body, join the game, or server restart?), would most likely make this more cohesive and more opt-in rather than an opt out system, or, god forbid she says this, a very forced system that nobody likes and people leave over. Some people might as well have jumped the gun and assumed that it was going to be a permanent change rather than just a simple testmerge - and with how Deadly's said all he's done is share a link to those MHelps asking about the right click menu (please don't do this, actually talk to people), it most likely _has_ made it feel like an unwelcoming mess.
> > TL;DR: A change that a lot of people neither expected, nor liked, getting dropped on people without asking for pre-project criticism, with people being unhelpful in both regards (people leaving _and_ staff not being communicative other than just links or lack of responses), can end up becoming a disaster for those very used to the old system - equally so for new players as now they have yet another thing to learn that isn't easy to understand buttons and their names of what they do.
> 
> This is a very long form way to say that you personally don’t like the system. Threatening players leaving isn’t going to change the minds of people who think this is a good mechanical change. Also, TG and monke both use combat mode… so unless you’re coming from goon, they’re the same.

What.
What makes you jump to that conclusion? Also, it wasn't threatening - she stated it more as a possibility (which most likely has happened, people leave for a variety of reasons and this may be a larger reason than others previously given).

Alongside that, the PR, as others have stated (mainly on the Discord), feels more like "a change for the sake of change." Also, yes, it screws with basic instincts (and habits). Now right click is no longer right click and you can't figure out how to right click, so now you're screwed because you can't figure it out on the spot and now you're **DEAD** because someone had the advantage of... having come by first and learned the new system that they probably also dislike.

...okay, that's just a scenario for the sake of expressing how this change-for-the-sake-of-change change can end up bad for any returning players who may or may not be in the discord, and/or may not have read the announcement. Before then getting flashbanged mentally by this new system that they most likely haven't seen yet, and were trying to go around normally because they didn't need to punch or shove yet, as suddenly they're in an unarmed fight that had appeared suddenly with an armed traitor.

Admittedly, this can be chalked up to more overexaggeration, but it's Space Station 13, and you can't blame her for having something like that happen - except it was the quad-intent system (the old one) and not the bi-intent one (the PR).

But her original points still stand... alongside the new ones.

**Comment by Moltijoe** [comment]
> Alongside that, the PR, as others have stated (mainly on the Discord), feels more like "a change for the sake of change." Also, yes, it screws with basic instincts (and habits). Now right click is no longer right click and you can't figure out how to right click, so now you're screwed because you can't figure it out on the spot and now you're **DEAD** because someone had the advantage of... having come by first and learned the new system that they probably also dislike.

hate to break it to you, sarah has said on numerous occasions that she's porting combat mode because she wanted to implement active blocking, but needed a "secondary fire" button for it to be possible.
it's not just "change for the sake of change"

also, you are making quite a lot of assumptions about others opinions, even the short time i used combat mode it felt way better. The resist against them seems almost entirely down to "i'm used to this old thing and hate that it's being changed" rather than there being something implicitly wrong with combat mode.

**Comment by Cartlord** [comment]
yeah i've tried to put thought into it and get over the mindset of "new thing bad, old thing good"
the only real issue i'm holding onto is that i'm opposed to obfuscation of the right-click menu, and i already made a post about that here
i respect the intent behind this & the work that went into it, and i hope that more actually comes of this instead of it becoming the next modular computers

**Comment by Addust** [comment]
> > you fucked right click pickup you fool
> 
> If you mean the right click menu, that's shift + right click now.

no i can't pick up shit with that menu

**Comment by iamthesilly** [comment]
> no i can't pick up shit with that menu

Already fixed this as of the last commit

**Comment by Addust** [comment]
yep

**Comment by ofscarletdreams** [comment]
is this going to be made a toggle or am i to cry about it

**Comment by Moltijoe** [comment]
> is this going to be made a toggle or am i to cry about it

adding a toggle isn't maintainable considering how few coders we have

**Comment by AMyriad** [comment]
arguments

**Comment by ToasterBiome** [comment]
> yeah i've tried to put thought into it and get over the mindset of "new thing bad, old thing good" the only real issue i'm holding onto is that i'm opposed to obfuscation of the right-click menu, and i already made a post about that here i respect the intent behind this & the work that went into it, and i hope that more actually comes of this instead of it becoming the next modular computers

Unfortunately I would love if yogs could give everyone a third button on their mouse but some concessions would have to be given to fit this feature in. obviously in a perfect world it would mesh perfectly but it unfortunately isn't so. shift clicking is honestly, pretty easy to get used to once you realize that it was changed, in the same way that examining things in shift click or pointing is second nature at this point.

**Comment by cuackles** [comment]
image

deleting this was intended or accident ?

**Review by ToasterBiome — APPROVED** [review]
